semeio/communication/semeio_script.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+import logging
+from logging.handlers import BufferingHandler
+import os
+from types import MethodType
+from res.enkf import ErtScript
+from semeio.communication.reporter import FileReporter
+
+
+class _LogHandlerContext(object):
+    def __init__(self, log, handler):
+        self._log = log
+        self._handler = handler
+
+    def __enter__(self):
+        self._log.addHandler(self._handler)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._log.removeHandler(self._handler)
+
+
+class _ReportHandler(BufferingHandler):
+    def __init__(self, output_dir):
+        super(_ReportHandler, self).__init__(1)
+        self._reporter = FileReporter(output_dir)
+        self._namespace = "log.txt"
+
+    def flush(self):
+        for log_record in self.buffer:
+            self._reporter.publish_msg(self._namespace, self._format_record(log_record))
+
+        super(_ReportHandler, self).flush()
+
+    def _format_record(self, log_record):
+        log_fmt = "{log_level} [{log_time}]: {log_message}"
+        return log_fmt.format(
+            log_level=log_record.levelname,
+            log_time=datetime.datetime.fromtimestamp(log_record.created),
+            log_message=log_record.message,
+        )
+
+
+class SemeioScript(ErtScript):  # pylint: disable=too-few-public-methods
```

Comment:
I thinks this class deserves a bit of documentation, as it is quite convoluted with the internal function, report handler, and log handler context

Reply:
I agree 👍

Reply:
Wrote some documentation and split __init__ into some smaller functions